### DISCUSSION TITLE
Issue with compiling on WSL with old libmesh version

### DISCUSSION THREAD

**Discussion by louisBS** [idaholab/moose#18407] [Q&A Getting Started] [answered]
Hi all, 

I'm trying to install MOOSE on a Windows laptop. I've followed the "Getting Started" guide on WSL and Conda MOOSE environment installation. To compile my code I have to use an old version of libmesh (2021.01.14 is the last one that works without any conflict to resolve). So I tried to roll back to that version but I have the following warning when doing so: 

Verifying transaction: -
SafetyError: The package for moose-libmesh located at /home/louis/miniconda3/pkgs/moose-libmesh-2021.01.14-build_1
appears to be corrupted. The path 'libmesh/include/libmesh/fparser.hh'
has an incorrect size.
  reported size: 8484 bytes
  actual size: 8192 bytes

SafetyError: The package for moose-libmesh located at /home/louis/miniconda3/pkgs/moose-libmesh-2021.01.14-build_1
appears to be corrupted. The path 'libmesh/include/libmesh/fparser_ad.hh'
has an incorrect size.
  reported size: 7394 bytes
  actual size: 4096 bytes

When I try to compile after that I get all sorts of errors with FunctionParser. Do you have any idea of what's wrong? I must add that I already tried to uninstall and reinstall the conda moose environment but I still run into the same issue. Another thing that might be related: the first time I tried to downgrade libmesh to 2021.01.14, I did not see the aforementioned error message and everything went fine during compiling (no error message about FunctionParser) until my computer crashed in the middle of the process...

Thank you,
Louis

**Comment by milljm** [comment, marked as answer]
I think every issue you are experiencing is due to our recent moose-libmesh dependency problem.

The details are available here, as well as the solution: https://github.com/idaholab/moose/discussions/18378 to that specific issue.

I would ask that you simply "Try Again" by following the advice given in the aforementioned link. Basically, the steps will be:

```bash
conda deactivate
conda clean --all       # <-- adding this specifically in your case
conda env remove -n moose
conda create -n moose moose-libmesh moose-tools
conda activate moose
```

I added an additional step, as you do seem to possibly have a corrupted tarball package. I am not sure how _that_ happened. But they fix is simple.

Also note: If `conda install` takes forever to solve (or begin), I would suggest using `mamba`. The details of which are also in the link above.

**Reply by louisBS** [reply]
Thank you Jason I will try that :)

**Reply by louisBS** [reply]
So I tried the steps you suggested, including the conda clean --all step, however, I still have the same corrupted package issue when downloading the 2021.01.14 version of moose-libmesh, leading to the same Function Parser related errors when compiling. Do you have any idea of what the issue might be?

**Reply by milljm** [reply]
I do not know what is causing the package to become corrupt... We have many machines which are downloading the very same package without error.

I don't think the package itself is corrupt, but lets test that. Can you curl the file manually, and perform an shasum on it?

```bash
curl -L -O https://anaconda.org/idaholab/moose-libmesh/2021.07.14/download/linux-64/moose-libmesh-2021.07.14-build_0.tar.bz2

# Your SHA should match mine:
sha256sum moose-libmesh-2021.07.14-build_0.tar.bz2 
4da68980e558030de93d658f2ac7ac5ead3f1c2892199f8f172898daa381a1f1  moose-libmesh-2021.07.14-build_0.tar.bz2
```

If the above failed, we are standing up a new server to host our moose packages, if you would like to give that a try:
```bash
conda deactivate
conda config --remove channels idaholab
conda config --add channels https://conda.software.inl.gov/public
conda config --env --set ssl_verify false          # <-- we don't have a proper SSL cert in place just yet
conda create -n new-moose moose-libmesh
```
Just installing moose-libmesh, to quickly test a corrupt package...

Is your MOOSE repo up-to-date?

```bash
cd moose
git fetch origin
git rebase origin/master
```

Other than the above I don't understand how the file is becoming corrupt. When you're finished troubleshooting, you may want to re-enable SSL:
```bash
conda config --remove channels https://conda.software.inl.gov/public
conda config --add channels idaholab
conda config --env --set ssl_verify true
```

**Reply by hugary1995** [reply]
@milljm I think he is trying to download an older version of libMesh from January.

@louisBS can you tell us your moose commit hash?

**Reply by milljm** [reply]
@hugary1995 you're right, I didn't see that. I suppose then, why is Conda doing that? The latest available packages have no dependency failures. The steps I outlined in my first response should have worked.

This statement sorta threw me for a loop:

_To compile my code I have to use an old version of libmesh (2021.01.14 is the last one that works without any conflict to resolve)_

I thought the conflict reference was to an actual conflict being caused by Conda packages. Which did happen recently.

So, @louisBS you are purposefully installing an older moose-libmesh and getting a corrupted download? I tried the below, but received no errors after installation. Admittedly I am using mamba, not conda, as I no longer have a stomach for the slow solve times (`conda install mamba`).

```bash
$> mamba create -n testing moose-libmesh=2021.01.14 moose-tools
<trimmed>
Finished moose-libmesh                        (00m:17s)              52 MB      3 MB/s
Downloading  [================================================] (00m:50s)   23.22 MB/s
Extracting   [================================================] (00m:50s)      58 / 58
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate testing
#
# To deactivate an active environment, use
#
#     $ conda deactivate
```

**Reply by louisBS** [reply]
@milljm @hugary1995 Yes, sorry for being unclear, I am indeed purposefully installing an older moose-libmesh. I just solved my problem by installing the same 2021.01.14 version of moose-libmesh but with mamba instead of conda as you suggested. It doesn't make much sense to me but I'm happy it works. Now my code compiles and runs just fine! Thank you for your time and answers guys.

If that can be of any interest for future troubleshooting, as I mentioned in my first message, the first time I tried to install libmesh 2021.01.14 a few days ago with conda it went fine (no message about corrupted packages) but my computer crashed while compiling. It's only after restarting my computer and trying to compile again that I got errors when compiling, as well as the message about corrupted package when trying to re-install the 2021.01.14 libmesh. As I just said, using mamba install instead of conda install solved the corrupted package issue.

**Reply by milljm** [reply]
`mamba` is just full of surprises 🎉
